Server.py

```diff
@@ -0,0 +1,224 @@
+import errno
+import select
+import socket
+import traceback
+
+import base
+import Client
+import CustomExceptions
+import util
+
+CLOSE, SERVER, CLIENT = range(3)
+CRLF = "\r\n"
+END_HEARDER = 2 * CRLF
+
+
+class Server(base.Base):
+    _database = {}
+
+    def __init__(
+        self,
+        buff_size,
+        base_directory
+    ):
+        super(Server, self).__init__()
+        self._buff_size = buff_size
+        self._run = True
+        self.logger.info("Initialized server, buff size '%d'", buff_size)
+        self._base_directory = base_directory
+
+    def terminate(self):
+        self._run = False
+
+    def add_server(
```

Comment:
why do we care if it a server or a client? all it needs is an object to register.

we have to kind of objects, one is the listener and the other is the connection, the listener will receive onread then instantiate the connection while passing accepted socket and add to the poller.

your poller should not care what is registered as long as it provides socket and events.